src/EditModel/EditDataEntryForm/PaletteSection.component.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const filterItems = ({ keySet, filter }) => {
+    const keys = Object.keys(keySet)
+    if (!filter.length) {
+        return keys
+    }
+
+    return keys.filter(key => filter.every(
+        f => keySet[key].toLowerCase().includes(f.toLowerCase())
+    ));
+};
+
+const PaletteSection = ({
```

Comment:
This was extracted from https://github.com/dhis2/maintenance-app/blob/224f2ec40242776d1e92ab5d6fdc2cc9d4989c88/src/EditModel/EditDataEntryForm.component.js#L405-L443